apis/kueue/v1alpha1/tas_types.go

```diff
@@ -87,24 +87,18 @@ type TopologyLevel struct {
 	NodeLabel string `json:"nodeLabel"`
 }
 
-// TopologyStatus defines the observed state of Topology
-type TopologyStatus struct {
-}
-
 // +genclient
 // +genclient:nonNamespaced
 // +kubebuilder:object:root=true
 // +kubebuilder:storageversion
-// +kubebuilder:subresource:status
 // +kubebuilder:resource:scope=Cluster
 
 // Topology is the Schema for the topology API
 type Topology struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   TopologySpec   `json:"spec,omitempty"`
-	Status TopologyStatus `json:"status,omitempty"`
```

Comment:
Maybe instead add generic `conditions` as for cohorts: https://github.com/kubernetes-sigs/kueue/blob/384d51e616177cea79bbc9e5dea4a0d40fab099f/apis/kueue/v1alpha1/cohort_types.go#L72

Reply:
I discussed this with @mimowo 
But, we were not sure if we could remove this field now, then we can re-add the status field/subresource as a non-pointer field since the API document mention that we should make optional field pointer:

> "ensure the field is entirely absent from API responses when empty (optional fields must be pointers)

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version

Addtiaionally, the status document doesn't mention anything for the pointer: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

The key point is, can we get rid of this empty status field to avoid confusion? After that, can we add the status field as a non-pointer field?
@alculquicondor Do you have any knowledge?

Reply:
I have consulted this with @liggitt and @alculquicondor and no need for pointer if we decide to add it in the future.

 In that case it is better to don't add it prematurely.

/hold cancel

Reply:
Thank you for asking to experts!